ironfish/src/blockchain/database/blockchaindb.ts

```diff
@@ -411,4 +417,47 @@ export class BlockchainDB {
   ): Promise<SchemaValue<LeavesSchema<NoteEncryptedHash>>> {
     return this.notes.getLeaf(index, tx)
   }
+
+  async getNullifiersSize(tx?: IDatabaseTransaction): Promise<number> {
+    return this.nullifiers.size(tx)
+  }
+
+  async getTransactionHashByNullifier(
+    nullifier: Nullifier,
+    tx?: IDatabaseTransaction,
+  ): Promise<TransactionHash | undefined> {
+    return this.nullifiers.get(nullifier, tx)
+  }
+
+  async connectBlockToNullifiers(block: Block, tx?: IDatabaseTransaction): Promise<void> {
+    return this.nullifiers.connectBlock(block, tx)
+  }
+
+  async disconnectBlockFromNullifiers(block: Block, tx?: IDatabaseTransaction): Promise<void> {
+    return this.nullifiers.disconnectBlock(block, tx)
+  }
+
+  async hasNullifier(nullifier: Nullifier, tx?: IDatabaseTransaction): Promise<boolean> {
+    return this.nullifiers.contains(nullifier, tx)
```

Comment:
I think after this merges I want to reverse this relationship to make this the same as existing components.